lib/matplotlib/_api/__init__.py

```diff
@@ -189,6 +190,55 @@ def check_getitem(_mapping, **kwargs):
             .format(v, k, ', '.join(map(repr, mapping)))) from None
 
 
+def select_overload(funcs, *args, **kwargs):
+    """
+    Select the function signature that accepts ``*args, **kwargs``, and return
+    a mapping of bound arguments.
+
```

Comment:
The name is still pretty mysterious.  Maybe:
```suggestion
def select_function_sigature(funcs, *args, **kwargs):
    """
   Match a user-supplied set of arguments to a method (as defined in ``*args, **kwargs``)  with two or 
   more signatures, and return a mapping of bound arguments to the chosen signature.  Useful when 
   changing the signature of a method but keeping back compatibility; usually the back-compatible 
   signature will still be passed, but with a deprecation warning.
```

Reply:
Overloading is pretty standard terminology (https://en.wikipedia.org/wiki/Function_overloading).
The use case is explained quite in length in the Notes.

Reply:
If you are going to insist on jargon, we should at least need to make it google-able, so maybe `select_function_overload`?  "function overload" at least gets me to the right place, "overload" is too general.    No doubt I am more ignorant than most of our contributors, but having `select_overload` show up in the middle of the code does not tell me what I am looking at, whereas `select_function_signature` does.

Reply:
Fair enough re: googlability.
What about `select_matching_signature`?

Reply:
That would be great for me...

Reply:
select_matching_signature it is, then.